crates/diesel_models/src/business_profile.rs

```diff
@@ -103,25 +103,39 @@ impl From<BusinessProfileNew> for BusinessProfile {
 
 impl BusinessProfileUpdateInternal {
     pub fn apply_changeset(self, source: BusinessProfile) -> BusinessProfile {
+        let Self {
+            profile_name,
+            modified_at,
+            return_url,
+            enable_payment_response_hash,
+            payment_response_hash_key,
+            redirect_to_merchant_with_http_post,
+            webhook_details,
+            metadata,
+            routing_algorithm,
+            intent_fulfillment_time,
+            frm_routing_algorithm,
+            payout_routing_algorithm,
+            is_recon_enabled,
+            applepay_verified_domains,
+        } = self;
         BusinessProfile {
-            profile_name: self.profile_name.unwrap_or(source.profile_name),
-            modified_at: self.modified_at.unwrap_or(source.modified_at),
-            return_url: self.return_url,
-            enable_payment_response_hash: self
-                .enable_payment_response_hash
+            profile_name: profile_name.unwrap_or(source.profile_name),
+            modified_at: modified_at.unwrap_or(source.modified_at),
```

Comment:
Why are we updating `modified_at` from the changeset? The updating user should not be in control of this field.
```suggestion
            modified_at: common_utils::date_time::now(),
```